spec/controllers/songs_controller_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'rails_helper'
+
+RSpec.describe SongsController, type: :controller do
+  let(:valid_attributes) do
+    {
+      title: "Talisman",
+      artist_name: "Air",
+      release_year: 2007,
+      released: true,
+      genre: "Post-Rock"
+    }
+  end
+
+  let(:invalid_attributes) do
+    {
+      title: nil,
+      artist_name: nil,
+      release_year: nil,
+      released: true
+    }
+  end
+
+  describe "basic listing, viewing and editing" do
+    let(:song) { Song.create!(valid_attributes) }
+
+    it "lists all songs" do
+      get :index
+      expect(assigns(:songs)).to eq([song])
```

Comment:
What do you recommend here?

If we test the HTML, then we're duplicating `spec/views/songs/index.html.erb_spec.rb`.

If we test the ActiveRecord functions, then we're not actually testing the controller at all.